### PR TITLE
FAD-5609 Subaccount features in Templates

### PR DESCRIPTION
**Pull request by jonambas**
**Changes**
- Subaccount support to template actions / links
- New `SubaccountSection` to handle subaccount assignment in the form.
- Updated templates `ListPage` with `hasSubaccounts` and `SubaccountTag`
- Added `containers` to templates all template pages, and some components.
- `*/containers/*` and `*.container.js` ignored by test coverage.

**List**
screen shot 2018-01-22 at 4 34 28 pm

**Create**
screen shot 2018-01-22 at 4 34 03 pm

**Edit/Published Master**
screen shot 2018-01-22 at 4 33 36 pm

**Edit/Published Assigned**
screen shot 2018-01-22 at 4 33 53 pm

Actions to verify:
- [x] Create - master, shared, assigned. Sending domain list updates on subaccount change.
- [x] Update - share toggle, assigned / unassigned
- [x] Delete - assigned / unassigned
- [x] Publish - assigned / unassigned
- [x] Preview
- [x] Send

To do:
- [x] Write Tests
- [x] Verify duplicating works

### CODE REVIEW

File: src/pages/templates/components/SubaccountSection.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    if (assignTo !== 'subaccount' && prevProps.assignTo === 'subaccount') {
+      change(formName, 'subaccount', null);
+    }
+  }
+
+  renderCreate() {
+    const { assignTo } = this.props;
+
+    const typeahead = assignTo === 'subaccount'
+      ? <Field name='subaccount' component={SubaccountTypeaheadWrapper} />
```
Comment (jonambas) [inline src/pages/templates/components/SubaccountSection.js]: add `validate={required}`

**Comment by jasonrhodes** [comment]
This seems to work well for the subaccount typeahead, generally:
```diff
diff --git a/src/components/subaccountTypeahead/SubaccountTypeahead.js b/src/components/subaccountTypeahead/SubaccountTypeahead.js
index 3f8813c8..5feb4e31 100644
--- a/src/components/subaccountTypeahead/SubaccountTypeahead.js
+++ b/src/components/subaccountTypeahead/SubaccountTypeahead.js
@@ -29,7 +29,7 @@ export class SubaccountTypeahead extends Component {
     clearSelection,
     isOpen
   }) => {
-    const { name, subaccounts, disabled, label = 'Subaccount', placeholder = 'None', error } = this.props;
+    const { name, subaccounts, disabled, label = 'Subaccount', placeholder = (isOpen ? 'Type to search' : 'None'), error } = this.props;
```

File: src/pages/templates/containers/EditPage.container.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { reduxForm } from 'redux-form';
+
+import { getDraft, getPublished, update, deleteTemplate, publish, getTestData } from 'src/actions/templates';
+import { showAlert } from 'src/actions/globalAlert';
+import { selectTemplateById, selectTemplateTestData, selectSubaccountIdFromQuery } from 'src/selectors/templates';
```
Comment (jasonrhodes) [inline src/pages/templates/containers/EditPage.container.js]: Can we move this `selectSubaccountIdFromQuery` to the subaccount selector file, or all on its own? We will probably want to share that across all places.

File: src/pages/templates/containers/PreviewDraftPage.container.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { showAlert } from 'src/actions/globalAlert';
+import { getDraftAndPreview, sendPreview } from 'src/actions/templates';
+import { selectDraftTemplate, selectDraftTemplatePreview, selectSubaccountIdFromQuery } from 'src/selectors/templates';
+import { getSubaccountQuery } from 'src/helpers/templates';
```
Comment (jasonrhodes) [inline src/pages/templates/containers/PreviewDraftPage.container.js]: Can we move this helper to `src/helpers/subaccounts` or its own helper file in `src/helpers` so we can easily share?

File: src/helpers/subaccounts.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
-export function getSubaccountQuery(id) {
+export function setSubaccountQuery(id) {
```
Comment (jonambas) [inline src/helpers/subaccounts.js]: Renamed to be in line with #250

**Comment by jonambas** [comment]
Update MB to 1.8.17 - input helptext styles (no snapshot changes)

**Pull request merged**